Implement the debug endpoint

kahowell opened the pull request:
With this change the debug endpoint accurately reflects what would be sent to the inventory service.

Since we control the interface for it, I decided that ConsumerInventory should reflect our service's view of a consumer generally, and then realized that there was a great deal of overlap with ConduitFacts.

Since the validation provided by ConduitFacts has been valuable, I reworked it to be a subclass of ConsumerInventory.

With this in place, `curl localhost:8080/rhsm-conduit/v1/inventories/1234` (or similar) is useful.

awood commented on src/main/java/org/candlepin/insights/inventory/ConduitFacts.java:
```diff
@@ -20,149 +20,58 @@
 
-    public void setFqdn(String fqdn) {
-        this.fqdn = fqdn;
+    // This is a soft validation.  Bogus IP addresses like "999.999.999.999" will still validate
```
I don't know that this comment is accurate anymore!  @mstead used [InetAddresses isInetAddress](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/net/InetAddresses.html#isInetAddress(java.lang.String)) method and it returns false for addresses like "999.3.3.3" (which is in the IpAddressValidatorTest.testInvalidIpV4Ip unit test).

kahowell replied:
It certainly isn't. I sloppily pulled this back in from when I rebased my changes. Will address in a follow-up. Thanks for catching!

awood commented on src/main/java/org/candlepin/insights/resource/InventoriesResource.java:
```diff
@@ -43,20 +38,17 @@
 
-    @Autowired
-    private TaskManager tasks;
+    public InventoriesResource(InventoryController inventoryController, TaskManager tasks) {
```
How are these getting injected without the `Autowired` annotation?

kahowell replied:
Per [spring docs](https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-spring-beans-and-dependency-injection.html), if the bean has only one constructor, `@Autowired` can be omitted for constructor injection. I'm not opposed to adding `@Autowired` for clarity if you'd like though?

awood replied:
Interesting!  No need to add it.

kahowell commented:
@awood, removed that stale comment, and added a missing newline in InventoriesResourceTest I noticed while looking back through the changes (can see `git diff a45fe59..b7d03f6` if curious).
